gcc/rust/resolve/rust-ast-resolve-pattern.h

```diff
@@ -45,6 +45,30 @@ struct PatternBinding
   {}
 };
 
+// Info that gets stored in the map. Helps us detect if two bindings to the same
+// identifier have different mutability or ref states.
+struct BindingInfo
+{
+  bool is_mut;
```

Comment:
I think it would be ideal to reuse the Mutability enum from the rust/util/rust-common.h